### PR TITLE
add documentation for QChemDictSet

### PR DESCRIPTION
**Pull request by rkingsbury**
Add documentation to `QChemDictSet` and potentially other QChem `InputSet`

TODO: 
- [x] Clarify type of `pcm_dielectric`. The original docstring specified a `str` but I believe a `float` is more appropriate and intuitive
- [x] Copy docstring to child sets
- [x] complete documentation of `opt_variables` and `scan_variables`

### CODE REVIEW

**Comment by rkingsbury** [comment]
@samblau @espottesmith would love to get your feedback on this. As I started using the QChem workflows I found myself having to do a lot of code diving and searching the QChem manual to understand how to set things properly. That's what motivated me to do this; hoping to provide some clarification for new users about what typical settings are.

**Comment by rkingsbury** [comment]
@samblau @espottesmith friendly reminder - any comments on this revised docstring?

**Comment by samblau** [comment]
My apologies for the delay, Ryan. Hoping to get to this later today.

**Comment by samblau** [comment]
Hi @rkingsbury, thanks for your patience. These are very helpful additions. A few suggested tweaks:

- I haven't checked recently, but not that long ago the new GenSCFman module was limited in what SCF algorithms it supported. It's muuuuuuch better than the old SCF code across the board, so I'd say it's basically always better to use it even if it has limitations. Thus, I'd recommend changing some of the documentation for scf_algorithm to note that DIIS, DIIS_GDM, and GDM are definitely supported by GenSCFman, and that if the user wants to use a different SCF algorithm, they should consult the manual to see if it is supported by GenSCFman. If it isn't, they might have to set genscfman to False in rem - I'm not sure what is default at this point or if requesting an SCF alg that is only in the old module will automatically force the switch. 
- I would perhaps not use tetrahydrofuran as an example SMD solvent input. There is this super awkward thing (which I hope has been fixed, but I'm not sure) where you actually have to request "thf" instead of "tetrahydrofuran" in the QChem input file. There are two stupid lines of code in QCInputs to handle this, but yeah, just maybe not the best example to avoid any potential confusion.
- Beyond rem, pcm, and solvent, overwrite_inputs will also work with smx and plots as per the lines following "if self.overwrite_inputs:".

Thanks for these additions!

**Comment by rkingsbury** [comment]
> Use  DIIS  unless  performing  a  restricted  open-shell  calculation,  in  which  case  GDM  is  rec-ommended.   If  DIIS  fails  to  find  a  reasonable  approximate  solution  in  the  initial  iterations,RCA_DIIS is the recommended fallback option. If DIIS approaches the correct solution but failsto finally converge, DIIS_GDM is the recommended fallback.  For systems with small HOMO-LUMO gaps and DIIS fails to converge, LS_DIIS could help

File: pymatgen/io/qchem/sets.py [hunk 1]
```diff
@@ -521,18 +667,47 @@ def __init__(
...
-            custom_smd ():
-            opt_variables ():
-            scan_variables ():
-            scf_algorithm ():
-            max_scf_cycles ():
-            plot_cubes ():
-            overwrite_inputs ():
+            molecule (Pymatgen Molecule object)
+            opt_variables (str or list):
+            scan_variables (str or list):
```
Comment (rkingsbury) [inline pymatgen/io/qchem/sets.py:672]: @espottesmith can you give me an example of how these variables should be populated for a pes_scan job?

**Comment by samblau** [comment]
@rkingsbury The SCF docstring looks good now. I'm 100% confident that DIIS_GDM uses the new algorithm. I'm hoping that RCA_DIIS either already does or will soon. At some point I'll need to go back and check because if RCA_DIIS is now compatible with GenSCFman then it almost certainly has a place in our SCF error handling procedure. All the other minor stuff looks good as well. Thanks!

**Comment by rkingsbury** [comment]
@shyuep This is ready for review / merge at your convenience. For some reason the flake8 linter is failing with

```
BLK100 Black would make changes
```
even though I have run `black` on the file in question. If you want me to take any further action on the formatting, please let me know.

**Comment by rkingsbury** [comment]
@samblau I just realized that changing `pcm_dielectric` from string to int is causing a test failure. I can fix it, but want to confirm you agree that we should change the type of this variable first.

File: pymatgen/io/qchem/sets.py [hunk 2]
```diff
@@ -136,7 +173,7 @@ def __init__(
 
         if self.pcm_dielectric is not None:
             mypcm = pcm_defaults
-            mysolvent["dielectric"] = self.pcm_dielectric
+            mysolvent["dielectric"] = str(self.pcm_dielectric)
```
Comment (rkingsbury) [inline pymatgen/io/qchem/sets.py:176]: @samblau

**Comment by rkingsbury** [comment]
@shyuep This is ready for review / merge at your convenience. All tests are passing, except that for some reason the flake8 linter is failing with

BLK100 Black would make changes

even though I have run black on the file in question. If you want me to take any further action on the formatting, please let me know.

**Comment by shyuep** [comment]
This usually means you are not running black from pymatgen root directory.

**Comment by rkingsbury** [comment]
@shyuep Thanks for the guidance. I ran `black pymatgen/io/qchem` from the root directory, pushed the changes, but am still getting the same linting error. Any other ideas?

**Comment by rkingsbury** [comment]
@shyuep following up your comment about `Black would make changes` warnings in #2006 , this PR is ready to merge. Thanks.

**Comment by shyuep** [comment]
@rkingsbury For the record, when I ran black from my pymatgen root, I get

```
reformatted /Users/shyue/repos/pymatgen/pymatgen/io/qchem/utils.py
reformatted /Users/shyue/repos/pymatgen/pymatgen/io/qchem/sets.py
reformatted /Users/shyue/repos/pymatgen/pymatgen/io/qchem/inputs.py
reformatted /Users/shyue/repos/pymatgen/pymatgen/io/qchem/tests/test_sets.py
reformatted /Users/shyue/repos/pymatgen/pymatgen/io/tests/test_gaussian.py
reformatted /Users/shyue/repos/pymatgen/pymatgen/io/qchem/outputs.py
reformatted /Users/shyue/repos/pymatgen/pymatgen/io/vasp/tests/test_outputs.py
All done! ✨ 🍰 ✨
7 files reformatted, 485 files left unchanged.
```

Further, there were pylint errors.
```
************* Module pymatgen.io.qchem.outputs
pymatgen/io/qchem/outputs.py:700:15: R1726: Boolean condition 'parsed_geometry == [] or None' may be simplified to 'parsed_geometry == []' (simplifiable-condition)
pymatgen/io/qchem/outputs.py:721:11: R1726: Boolean condition 'parsed_optimized_geometries == [] or None' may be simplified to 'parsed_optimized_geometries == []' (simplifiable-condition)
-----------------------------------
Your code has been rated at 9.98/10
```

I really suggest you use the pre-commit hook included in pymatgen. That will avoid a lot of issues in future. I have no idea why black made changes in my case when it didn't in your case. But pylint errors should be dealt with regardless.

**Comment by shyuep** [comment]
@rkingsbury See this commit for the actual changes made by black: https://github.com/materialsproject/pymatgen/commit/d67a637825647cb8ecae95f07021be37f2bd8d7c

**Comment by rkingsbury** [comment]
Thanks @shyuep! I'll have to examine my local configuration and figure out what's going on. Can you tell me what version of black you're using? Somehow it seems that several of my local linter configuration has gone out of sync with upstream.

**Comment by shyuep** [comment]
I am using black==20.8b1. It turns out the flake8-black was using an old version on the Github actions. I just added a pin to black==20.8b1. Hopefully that solves the issue on Github Actions being inconsistent with our local machines. I also rearranged the linters such that the other lints are done first and flake8 is the last one to run.

**Pull request merged**